### PR TITLE
adding Binary Insertion Sort Algorithm JavaScript implementation

### PR DESCRIPTION
**Pull request by OussamaSALAHOUELHADJ**
## Assigned Issue Number?
#254

## Language Opted?
JavaScript

## Number of files added or updated?
1

## Name of these files?
Algorithms => Sorting => Binary_Insertion_Sort => binaryInsertionSort.js

### Checklist

- [x] I was assigned for this otherwise my **PR will be rejected**.
- [x] I've read the contribution guidelines.
- [x] I've written code by myself. **No Plagiarism allowed**
- [x] I've referred the correct issue number.
- [x] I've filled up this entire template correctly.

:star2: this repository and support.

### CODE REVIEW

File: Algorithms/Sorting/Binary_Insertion_Sort/binaryInsertionSort.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+//Swap function swaps two positions in an arra and returns the new array.
+function swap(array, firstIndex, secondIndex) {
+    let exchParam = array[firstIndex];
+    array[firstIndex] = array[secondIndex];
+    array[secondIndex] = exchParam;
+
+    return array;
+}
+
+function binarySort(array) {
```
Comment (hemanth-kotagiri) [inline Algorithms/Sorting/Binary_Insertion_Sort/binaryInsertionSort.js:10]: This is not a Binary Insertion Sort Algorithm.

**Review by hemanth-kotagiri — CHANGES_REQUESTED** [review]
Hey, @OussamaSALAHOUELHADJ. Refer to [this](https://en.wikipedia.org/wiki/Insertion_sort#Variants) link to learn about Binary Insertion sort and make necessary changes.

**Comment by OussamaSALAHOUELHADJ** [comment]
Hey @hemanth-kotagiri, Thank you so much for informing me about that, I was confused between Insertion sort algorithm and binary Insertion sort algorithm. My apologies. I have made the necessary changes and pushed the new code. 

This algorithm has been tested with the following script that I wrote:
```javascript
let times = 1;
let passed = true;

do {

    let arr = [];
    for (let index = 0; index < Math.floor(Math.random() * 10000); index++) {
        arr.push(Math.floor(Math.random() * 10000));
    }

    const clnArr = arr.slice(0);
    clnArr.sort((a, b) => a - b);

    const binarysortedarray = binaryInsertionSort(arr);

    for (let index = 0; index < arr.length; index++) {
        passed = clnArr[index] === binarysortedarray[index];
        if (!passed) {
            break
        }
    }

    passed ? console.log("Test case N " + times + " Passed ✅\n ------") : (console.log("Error on test case N" + times, arr), console.log("\nshould be\n", clnArr), console.log("\nbut Instead\n", binarysortedarray))
    times++
} while (passed && times <= 100000);
```

I tested the Algorithm over than 100000 in multiple times with a Passed ✅ every time.

```
⋮
 ------
Test case N 99997 Passed ✅
 ------
Test case N 99998 Passed ✅
 ------
Test case N 99999 Passed ✅
 ------
Test case N 100000 Passed ✅
 ------
```

By the way, adding test cases for the algorithms is a great idea to verify the code. 

Please verify the PR and let me know if any changes needed.

**Review by hemanth-kotagiri — APPROVED** [review]
LGTM.

**Pull request merged**